pkgs/development/tools/java/jprofiler/default.nix

```diff
@@ -0,0 +1,100 @@
+{ stdenv
+, lib
+, fetchurl
+, makeWrapper
+, makeDesktopItem
+, undmg
+, jdk11 }:
+
+let
+  inherit (stdenv.hostPlatform) system;
+  pname = "jprofiler";
+
+  # 11.1.4 is the last version which can be unpacked by undmg
+  # See: https://github.com/matthewbauer/undmg/issues/9
+  version = if stdenv.isLinux then "13.0.2" else "11.1.4";
+  nameApp = "JProfiler";
+
+  meta = with lib; {
+    description = "JProfiler's intuitive UI helps you resolve performance bottlenecks";
+    longDescription = ''
+      JProfiler's intuitive UI helps you resolve performance bottlenecks,
+      pin down memory leaks and understand threading issues.
+    '';
+    homepage = "https://www.ej-technologies.com/products/jprofiler/overview.html";
+    license = licenses.unfree;
+    maintainers = with maintainers; [ catap ];
+  };
+
+  src = fetchurl (if stdenv.isLinux then {
+    url = "https://download-gcdn.ej-technologies.com/jprofiler/jprofiler_linux_${builtins.replaceStrings ["."] ["_"]  version}.tar.gz";
+    sha256 = "sha256-x9I7l2ctquCqUymtlQpFXE6+u0Yg773qE6MvAxvCaEE=";
+  } else {
+    url = "https://download-gcdn.ej-technologies.com/jprofiler/jprofiler_macos_${builtins.replaceStrings ["."] ["_"]  version}.dmg";
+    sha256 = "sha256-WDMGrDsMdY1//WMHgr+/YKSxHWt6A1dD1Pd/MuDOaz8=";
+  });
+
+
+  srcIcon = fetchurl {
+    url = "https://www.ej-technologies.com/assets/content/header-product-jprofiler@2x-24bc4d84bd2a4eb641a5c8531758ff7c.png";
+    sha256 = "sha256-XUmuqhnNv7mZ3Gb4A0HLSlfiJd5xbCExVsw3hmXHeVE=";
+  };
+
+  desktopItem = makeDesktopItem {
+    name = pname;
+    exec = pname;
+    icon = pname;
+    comment = meta.description;
+    desktopName = nameApp;
+    genericName = "Java Profiler Tool";
+    categories = [ "Development" ];
+  };
+
+  linux = stdenv.mkDerivation {
+    inherit pname version src;
+
+    nativeBuildInputs = [ makeWrapper ];
+
+    installPhase = ''
+      cp -r . $out
+
+      rm -f $out/bin/updater
+
+      for f in $(find $out/bin -type f -executable); do
+        wrapProgram $f --set JAVA_HOME "${jdk11.home}"
+      done
+
+      install -Dm644 "${desktopItem}/share/applications/"* \
+        -t $out/share/applications/
+      install -Dm644 "${srcIcon}" \
+        "$out/share/icons/hicolor/scalable/apps/jprofiler.png"
+    '';
+
+    meta = meta // { platforms = lib.platforms.linux; };
+  };
+
+  darwin = stdenv.mkDerivation {
+    inherit pname version src;
+
+    # by some reason undmg can't open it with error
+    #   Couldn't open extents tree
+    #   error: could not get volume
+    nativeBuildInputs = [ makeWrapper undmg ];
+    # unpackPhase = ''
+    #   7z x ${src}
+    # '';
```

Comment:
Please don't leave commented out code, it will only confuse people. If you think that this must be kept, add a comment explaining why.
```suggestion
```